### DISCUSSION TITLE
Provide a way to link to a given standard name(s) in the table page

### DISCUSSION THREAD

**Discussion by sadielbartholomew** [cf-convention/discuss#344] [Comments and ideas for changing CF]
### Topic for discussion

We can reference the Standard Name full table via the URL ['https://cfconventions.org/Data/cf-standard-names/current/build/cf-standard-name-table.html'](https://cfconventions.org/Data/cf-standard-names/current/build/cf-standard-name-table.html), but I can't seem to find a way to do what I would like to do, namely to link to a given name within the table, much like when you link to a certain, say, section within a webpage often via an appended  `#<section identifier>` which opens the page centered on specific content (for example, the link https://cfconventions.org/Data/cf-conventions/cf-conventions-1.11/cf-conventions.html#_introduction` will open the CF Conventions document at the 'Introduction' section rather than from the start).

My specific use case now was to refer to a name that I am advising someone to use for their dataset to make it more compliant as opposed to a `ncvar%datetime` (here `forecast_reference_time`) and thought it would be nice to reference it _directly from/within the table_ for them. _I like this because it demonstrates a name you are referencing is an official name and also encourages the exploration of the table._

I propose we have a way to link to a search result query, so that we can link now only to one name but to a given set of names as well.

I think this should be viable code-wise, though we will need to utilise the search mechanics to enable related query parameters in the link, at least in the approach I am envisioning.

Thoughts would be appreciated. Thanks.

**Comment by DocOtak** [comment]
Maybe link directly to the concept in NVS? http://vocab.nerc.ac.uk/collection/P07/current/CFSN0622/

Those [URI fragments](https://en.wikipedia.org/wiki/URI_fragment) can point at any id in the document, for your example name, the closest id is the table row which has the id forecast_reference_time_tr: https://cfconventions.org/Data/cf-standard-names/current/build/cf-standard-name-table.html#forecast_reference_time_tr

**Reply by sadielbartholomew** [reply]
Aha, thanks Andrew the NVS link is a great way to reference a specific name. As a workaround for now, it is really useful.

That said, I still think it would be useful to reference the name from the table itself, if only to encourage exploration of the table as an additional benefit. Your URI fragment link is somewhat on the way there, but it doesn't highlight the name itself which is what I ideally want, and have it pre-selected so that the description appears, to point to it more clearly than just as the first element shown on the page. If there's a way with the URI fragment to do that, then my request is already solved, but otherwise ideally we can find a way to do that.


But it does depend quite how difficult it might be to set up that.

**Reply by DocOtak** [reply]
Text fragments might help: https://cfconventions.org/Data/cf-standard-names/current/build/cf-standard-name-table.html#%23:~:text=forecast_reference_time

Opening the details is done with javascript so that would probably require changing the page template.

**Comment by graybeal** [comment]
I was going to recommend the same thing, but from a different source that uses semantic fragments (for better or worse :->): 
  https://mmisw.org/ont/cf/parameter/air_pseudo_equivalent_temperature 

There has been longstanding interest from the semantic community ([see ticket](https://github.com/cf-convention/vocabularies/issues/11)) in CF minting its own namespace, or maybe I should say using its own namespace. It could follow the lead of SWEET for example, which uses content redirection to redirect to an existing resolution service. 
  http://sweetontology.net/humanKnowledgeDomain#biogeochemistry
Visiting http://sweetontology.net lands at a page that describes how to support this using other services, without having to provide your own resolution service at all.

Using a w3id.org namespace to provide the redirection is another resolution approach used by a lot of vocabularies, if you just can't decide what namespace your own vocabulary should live in.

**Reply by sadielbartholomew** [reply]
Thanks John - something like those, with separate pages resolved for each name and including all information for the given name, would be even better! I imagine those would take a bit more work, though.

Maybe getting something like this implemented, or moving towards that overall goal, could be a Hackathon topic for the upcoming, or a future, CF Workshop?

**Reply by DocOtak** [reply]
I would have interest in CF migrating away from hosting its own vocab. The NVS is so integral to the process already, I would support and perhaps even push for adopting it as the canonical location.

**Reply by sadielbartholomew** [reply]
I think Alison, @japamment, has been doing lots of work in the past years at least on the vocabulary side, and maybe there are plans in this respect, or maybe there are good reasons not to want to go for an externally-hosted vocab?

**Comment by JonathanGregory** [comment]
[@DocOtak wrote](https://github.com/orgs/cf-convention/discussions/344#discussioncomment-10321186)

> I would have interest in CF migrating away from hosting its own vocab. The NVS is so integral to the process already, I would support and perhaps even push for adopting it as the canonical location.

Alison proposed in January that the NERC vocabulary server [NVS](https://vocab.nerc.ac.uk) should be the canonical location for standard names, although CF would keep copies in their present format on our website, but they would be secondary. The CF committee all agreed with this idea, when we discussed it at the January meeting. I believe that we are expecting Alison to make such a proposal - but now the cat is out of the bag anyway, since I've said it. (Maybe it was already on the loose, and I've forgotten.) Would anyone object to that change?

**Comment by taylor13** [comment]
Sounds o.k., but have we considered what happens if NERC support dries up?  I assume we would have to transfer the "vocabulary reference source" somewhere else.  I recall transferring the management of the mail service from NCAR to PCMDI years ago was a real pain.  Would this not be a problem with NVS because CF would "keep a copy" on our CF website?  Just checking.

**Comment by JonathanGregory** [comment]
If I remember correctly, Alison's intention is that the CEDA vocabulary editor will remain the source, which CEDA will use to generate the content for the NERC vocabulary server as well as the CF website. Designating the NVS as the primary repository for the standard names is a formal change, in a sense, but does not change the process.

I hope I am not misrepresenting Alison. My memory may be mistaken. Perhaps @japamment @feggleton @efisher008 could comment?

**Comment by efisher008** [comment]
Dear @DocOtak, @JonathanGregory, and others,

Alison @japamment is on leave this week and will return on Monday 19th August. I would not want to misrepresent what she has previously suggested or recommended, so I would prefer to wait for her response in her own words (which she will be able to give next week). Thank you for your patience.

Best regards,
Ellie
